workflow/util/util.go

```diff
@@ -505,10 +505,10 @@ func FormulateResubmitWorkflow(wf *wfv1.Workflow, memoized bool) (*wfv1.Workflow
 	// Append an additional label so it's easy for user to see the
 	// name of the original workflow that has been resubmitted.
 	newWF.ObjectMeta.Labels[common.LabelKeyPreviousWorkflowName] = wf.ObjectMeta.Name
+	if newWF.ObjectMeta.Annotations == nil {
+		newWF.ObjectMeta.Annotations = make(map[string]string)
+	}
 	for key, val := range wf.ObjectMeta.Annotations {
-		if newWF.ObjectMeta.Annotations == nil {
```

Comment:
Hmm.. this way we don't create the map if `wf.ObjectMeta.Annotations` is empty or `nil` at the cost of a simple nil check, which is very cheap.

Reply:
But for large workflows there might be large number of annotations and this becomes non-trivial.

Reply:
I don't see this becoming a real issue, however I'll merge this for consistency with a similar change in https://github.com/argoproj/argo/pull/3261